test/distributed/_shard/test_replicated_tensor.py

```diff
@@ -74,3 +76,109 @@ def test_replicated_tensor_inter_op_tensor(self):
         self.assertNotIsInstance(new_tensor, ReplicatedTensor)
 
         self.assertEqual(new_tensor, local_tensor + local_rand_tensor)
+
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_replicated_tensor_inter_op_sharded_tensor(self):
+        local_tensor = torch.ones(3, 3, device=f"cuda:{self.rank}") * 4
+        replica_tensor = ReplicatedTensor(local_tensor)
```

Comment:
So I think we should do the following to ensure our assumptions around broadcasting with ReplicatedTensor are valid:

1. Create two local random tensors (t1, t2) and run all of these ops on them.
2. Now create a ShardedTensor from t1 and ReplicatedTensor from t2 and run the same ops.
3. Gather the results (ex: on rank 0) and validate to a local ops.